File: ambeth/Ambeth.Merge/ambeth/merge/IEntityFactoryExtension.cs
using De.Osthus.Ambeth.Merge.Model;
using System;

namespace De.Osthus.Ambeth.Merge
{
    public interface IEntityFactoryExtension
    {
        Type GetMappedEntityType(Type type);

        Object PostProcessMappedEntity(Type originalType, IEntityMetaData metaData, Object mappedEntity);
    }
}

File: jambeth/jambeth-merge/src/main/java/de/osthus/ambeth/merge/IEntityFactoryExtension.java
package de.osthus.ambeth.merge;

import de.osthus.ambeth.merge.model.IEntityMetaData;

public interface IEntityFactoryExtension
{
	<T> Class<? extends T> getMappedEntityType(Class<T> type);

	Object postProcessMappedEntity(Class<?> originalType, IEntityMetaData metaData, Object mappedEntity);
}
